[Feature]: Support block manager v2 for chunked prefill

### 🚀 The feature, motivation and pitch

Chunked prefill currently doesn't work with block manager v2. Add `use_v2_block_manager=True` to https://github.com/vllm-project/vllm/blob/main/tests/basic_correctness/test_chunked_prefill.py#L48 to reproduce the error:

```
    def append_token_ids(self, block_index: int, token_ids: List[int]) -> None:
>       block = self._blocks[block_index]
E       IndexError: list index out of range
```

### Alternatives

Use block manager v1

### Additional context

cc @rkooo567 @cadedaniel

--- a/tests/core/block/e2e/test_correctness.py
+++ b/tests/core/block/e2e/test_correctness.py
@@ -263,23 +263,20 @@ def test_lookahead_greedy_equality_with_preemption(baseline_llm_generator,
             "enable_chunked_prefill": True,
         },
     ])
-@pytest.mark.parametrize("per_test_common_llm_kwargs", [
-{
-    "block_size": 8,
-    "max_num_batched_tokens": 2,
-    "max_num_seqs": 2,
-},
-{
-    "block_size": 8,
-    "max_num_batched_tokens": 3,
-    "max_num_seqs": 2,
-},
-{
-    "block_size": 8,
-    "max_num_batched_tokens": 256,
-    "max_num_seqs": 10,
-}
-])
+@pytest.mark.parametrize("per_test_common_llm_kwargs",
+                         [{
+                             "block_size": 8,
+                             "max_num_batched_tokens": 2,
+                             "max_num_seqs": 2,
+                         }, {
+                             "block_size": 8,
+                             "max_num_batched_tokens": 3,
+                             "max_num_seqs": 2,
+                         }, {
+                             "block_size": 8,
+                             "max_num_batched_tokens": 256,
+                             "max_num_seqs": 10,
+                         }])
 @pytest.mark.parametrize("baseline_llm_kwargs", [
     {
         "use_v2_block_manager": False,
@@ -308,7 +305,7 @@ def test_chunked_prefill_block_manager_v2(baseline_llm_generator,
     prompts = [
         "Hello, my name is",
         "The president of the United States is",
-        ("1 + " * 50) + " 1 = ", # Longer prompt.
+        ("1 + " * 50) + " 1 = ",  # Longer prompt.
         "The capital of France is",
         "The future of AI is",
     ]

--- a/tests/core/block/e2e/test_correctness.py
+++ b/tests/core/block/e2e/test_correctness.py
@@ -261,11 +261,25 @@ def test_lookahead_greedy_equality_with_preemption(baseline_llm_generator,
             # skip cuda graph creation for fast test.
             "enforce_eager": True,
             "enable_chunked_prefill": True,
-            "max_num_batched_tokens": 2,
-            "max_num_seqs": 2,
         },
     ])
-@pytest.mark.parametrize("per_test_common_llm_kwargs", [{}])
+@pytest.mark.parametrize("per_test_common_llm_kwargs", [
+{
+    "block_size": 8,
+    "max_num_batched_tokens": 2,
+    "max_num_seqs": 2,
+},
+{
+    "block_size": 8,
+    "max_num_batched_tokens": 3,
+    "max_num_seqs": 2,
+},
+{
+    "block_size": 8,
+    "max_num_batched_tokens": 256,
+    "max_num_seqs": 10,
+}
+])
 @pytest.mark.parametrize("baseline_llm_kwargs", [
     {
         "use_v2_block_manager": False,
@@ -294,6 +308,7 @@ def test_chunked_prefill_block_manager_v2(baseline_llm_generator,
     prompts = [
         "Hello, my name is",
         "The president of the United States is",
+        ("1 + " * 50) + " 1 = ", # Longer prompt.
         "The capital of France is",
         "The future of AI is",
     ]

--- a/vllm/core/block/block_table.py
+++ b/vllm/core/block/block_table.py
@@ -356,7 +356,13 @@ class BlockTable:
         appended to blocks. The first such "token block" may have less token ids
         than the block size, since the last allocated block may be partially
         full.
+
+        If no token ids are provided, then no chunks are returned.
         """
+
+        if not token_ids:
+            return []
+
         first_chunk_size = self._block_size - (self._num_full_slots %
                                                self._block_size)
         token_blocks = [token_ids[:first_chunk_size]]

